src/NativeScript/ObjC/Unmanaged/UnmanagedPrototype.mm

```diff
@@ -0,0 +1,50 @@
+#include "UnmanagedPrototype.h"
+#include "UnmanagedInstance.h"
+#include "ObjCTypes.h"
+#include "ObjCConstructorBase.h"
+
+namespace NativeScript {
+static char consumedUnmanagedCheck = 'k';
+
+using namespace JSC;
+
+const ClassInfo UnmanagedPrototype::s_info = { "Unmanaged", &Base::s_info, 0, CREATE_METHOD_TABLE(UnmanagedPrototype) };
+
+static EncodedJSValue JSC_HOST_CALL takeRetainedValue(ExecState* execState) {
+    UnmanagedInstance* instance = jsDynamicCast<UnmanagedInstance*>(execState->thisValue());
+    if (instance->data() == &consumedUnmanagedCheck) {
+        return throwVMTypeError(execState, WTF::ASCIILiteral("Unmanaged value has already been consumed. "));
+    }
+
+    id result = static_cast<id>(instance->data());
+    instance->setData(&consumedUnmanagedCheck);
```

Comment:
That was my idea at first but then @fealebenpae suggested this approach as a cooler way to do it, so I decided to follow his advice :) Plus you don't need to add another member to the header. If we all agree on your suggestion, I will change it :)